fmt might break inclusive range syntax with floating point literals

```rust
#![feature(inclusive_range_syntax)]

fn main() {
    let x = [0. ...10.0];
}
```

gets formatted to

```rust
#![feature(inclusive_range_syntax)]

fn main() {
    let x = [0....10.0];
}
```

yielding the error:

```
rustc 1.20.0-nightly (067971139 2017-07-02)
error[E0586]: inclusive range with no end
 --> <anon>:4:18
  |
4 |     let x = [0....10.0];
  |                  ^
  |
  = help: inclusive ranges must be bounded at the end (`...b` or `a...b`)
```

--- a/src/expr.rs
+++ b/src/expr.rs
@@ -268,10 +268,26 @@ pub fn format_expr(
                 ast::RangeLimits::Closed => "...",
             };
 
+            fn needs_space_before_range(context: &RewriteContext, lhs: &ast::Expr) -> bool {
+                match lhs.node {
+                    ast::ExprKind::Lit(ref lit) => {
+                        match lit.node {
+                            ast::LitKind::FloatUnsuffixed(..) => {
+                                context.snippet(lit.span).ends_with('.')
+                            }
+                            _ => false,
+                        }
+                    }
+                    _ => false,
+                }
+            }
+
             match (lhs.as_ref().map(|x| &**x), rhs.as_ref().map(|x| &**x)) {
                 (Some(ref lhs), Some(ref rhs)) => {
                     let sp_delim = if context.config.spaces_around_ranges() {
                         format!(" {} ", delim)
+                    } else if needs_space_before_range(context, lhs) {
+                        format!(" {}", delim)
                     } else {
                         delim.into()
                     };

--- a/tests/source/expr.rs
+++ b/tests/source/expr.rs
@@ -251,6 +251,10 @@ fn ranges() {
     let y = aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ... bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb;
     let z = ... x ;
 
+    // #1766
+    let x = [0. ..10.0];
+    let x = [0. ...10.0];
+
     a ... b
 
     // the expr below won't compile for some reason...

--- a/tests/target/expr.rs
+++ b/tests/target/expr.rs
@@ -315,6 +315,10 @@ fn ranges() {
         bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb;
     let z = ...x;
 
+    // #1766
+    let x = [0. ..10.0];
+    let x = [0. ...10.0];
+
     a...b
 
     // the expr below won't compile for some reason...

